src/HttpLayer.hs

```diff
@@ -89,4 +112,8 @@ apiCallSafe parser req = do
     v <- getResponseBody <$> httpJSON req
     pure $ parseEither parser v
 
-
+-- | The mapLeft function takes a function and applies it to an Either value
+-- iff the value takes the form Left _.
+mapLeft :: (a -> c) -> Either a b -> Either c b
```

Comment:
You don't need this. You can use the function `either`.

Reply:
These two functions have similar, though different signatures:

```haskell
either :: (a -> c) -> (b -> c) -> Either a b -> c
mapLeft :: (a -> c) -> Either a b -> Either c b
```

The `mapLeft` function returns an `Either`, while `either` returns a plain value that erases the information if it is a valid or invalid value.

Reply:
In most cases, you don't need such functions, if you need to collapse either you use the `either` function, otherwise you can case match. Introducing functions for common operations just brings confusion.